### PR TITLE
Use GCP dry run function for generating Airflow DAGs

### PR DESCRIPTION
**Pull request by scholtzan**
Fixes https://github.com/mozilla/bigquery-etl/issues/1169

### CODE REVIEW

**Comment by scholtzan** [comment]
This depends on https://github.com/mozilla-services/cloudops-infra/pull/2406 getting deployed

File: tests/query_scheduling/test_dag_collection.py [hunk 1]
```diff
@@ -3,13 +3,18 @@
...
 from bigquery_etl.query_scheduling.dag_collection import DagCollection
 from bigquery_etl.query_scheduling.dag import InvalidDag, DagParseException
 from bigquery_etl.query_scheduling.task import Task
 from bigquery_etl.metadata.parse_metadata import Metadata
+from bigquery_etl.dryrun import DryRun
 
 TEST_DIR = Path(__file__).parent.parent
+TEST_DRY_RUN_URL = (
+    "https://us-central1-bigquery-etl-integration-test.cloudfunctions.net/dryrun"
+)
```
Comment (scholtzan) [inline tests/query_scheduling/test_dag_collection.py:17]: I created a dry run function in the `bigquery-etl-integration-test` which is used for the tests here. It's probably also useful to have for testing if changes are made to the function itself.

File: bigquery_etl/dryrun.py [hunk 2]
```diff
@@ -111,52 +107,100 @@
...
+    )
+
+    def __init__(self, sqlfile):
+        """Instantiate DryRun class."""
+        self.sqlfile = sqlfile
+        self.dry_run_result = None
+
+    def _execute(self):
+        """Dry run the provided SQL file."""
+        if self.dry_run_result:
```
Comment (jklukas) [inline bigquery_etl/dryrun.py:125]: I can't remember if the base python version here is 3.7 or 3.8. If 3.8, dry_run_result seems like a prime candidate to be expressed as a [cached property](https://docs.python.org/dev/library/functools.html#functools.cached_property), eliminating the need for `_execute`

File: tests/query_scheduling/test_dag_collection.py (hunk 1 above)
Comment (jklukas) [reply tests/query_scheduling/test_dag_collection.py:17]: How is that function managed? It seems potentially problematic that the function deployed in the test project could be outdated compared to the shared-prod version.

Comment (scholtzan) [reply tests/query_scheduling/test_dag_collection.py:17]: Currently, it's been created manually in `bigquery-etl-integration-test`. But maybe there is a way to deploy the Cloud Function in https://github.com/scholtzan/cloudops-infra/tree/master/projects/data-shared/tf/modules/cloudfunctions/src/bigquery_etl_dryrun to both projects?

Comment (jklukas) [reply tests/query_scheduling/test_dag_collection.py:17]: I think it's sufficient for now to put a comment in the code here about how this is currently deployed manually and could get out of date with what's deployed in shared-prod.

**Pull request merged**